src/repository/artefact.rs

```diff
@@ -494,7 +497,7 @@ impl Artefact for Binding {
             if let Some(inst) = src.instance() {
                 let mut instance = String::new();
                 for (map_name, map_replace) in &mappings {
-                    instance = inst.replace(&format!("%7B{}%7D", map_name), map_replace.as_str());
+                    instance = inst.replace(&format!("%7B{}%7D", map_name), &map_replace);
```

Comment:
literally having struggled with `.as_str()` for exactly three PR's ever, this is still awesome 😄

Reply:
Ja those things keep changing and depending on what is done different syntaxes are needed there is `as_str`, `as_ref`, `deref`, `borrow`, `&` that all do slightly different but related things :D.